server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -650,6 +665,13 @@ boolean isPatternMatchingAllIndices(Metadata metadata, String[] indicesOrAliases
         return false;
     }
 
+    private boolean isSystemIndexAccessAllowed() {
+        if (threadContext == null) {
```

Comment:
why would the threadContext be null?

Reply:
I think this was a hack I put in during development to work around some (since otherwise fixed) test issues that accidentally got committed. I'll remove it, thanks for the catch.